.github/workflows/release_artifacts.yml

```diff
@@ -44,3 +43,31 @@ jobs:
           -H "Content-Type: application/zip" \
           --data-binary @./tigerbeetle-${{ matrix.target }}-${GIT_TAG}${{ matrix.debug }}.zip \
           "https://uploads.github.com/repos/tigerbeetledb/tigerbeetle/releases/$RELEASE_ID/assets?name=tigerbeetle-${{ matrix.target }}-${GIT_TAG}${{ matrix.debug }}.zip"
+
```

Comment:
@eatonphil since we don't currently have de-dupe / caching of jobs, I'm worried about the extra jobs this will create since it'll essentially run the full pipeline again on main.

Reply:
Any suggestions?

Reply:
I've added more restrictions

Reply:
Great!